go/5_auth-gcp.go

```diff
@@ -31,34 +31,28 @@ func getSecretWithGCPAuthIAM() (string, error) {
 		return "", fmt.Errorf("unable to initialize Vault client: %w", err)
 	}
 
-	// Learn about authenticating to GCS with service account credentials at https://cloud.google.com/docs/authentication/production
-	if pathToCreds := os.Getenv("GOOGLE_APPLICATION_CREDENTIALS"); pathToCreds == "" {
-		fmt.Printf("WARNING: Environment variable GOOGLE_APPLICATION_CREDENTIALS was not set. IAM client for JWT signing and Vault server IAM client will both fall back to default instance credentials.\n")
-	}
+	svcAccountEmail := fmt.Sprintf("%s@%s.iam.gserviceaccount.com", os.Getenv("GCP_SERVICE_ACCOUNT_NAME"), os.Getenv("GOOGLE_CLOUD_PROJECT"))
 
-	jwtResp, err := signJWT()
+	// For IAM-style auth, the environment variable GOOGLE_APPLICATION_CREDENTIALS
```

Comment:
I added a warning null check here for `GOOGLE_APPLICATION_CREDENTIALS` in the dotnet example, that way if someone runs it they can realize quickly that it isn't set in this case.

Reply:
Oh right, I used to have that in mine too. I'll add it back in!